source/ClientRequest/AMDSClientContinuousDataRequest.h

```diff
@@ -10,27 +11,69 @@ class AMDSClientContinuousDataRequest : public AMDSClientDataRequest
 Q_OBJECT
 public:
 	Q_INVOKABLE explicit AMDSClientContinuousDataRequest(QObject *parent = 0);
-	explicit AMDSClientContinuousDataRequest(ResponseType responseType, const QString &socketKey, const QString &bufferName, bool includeStatusData, const QDateTime &startTime, const AMDSBufferGroupInfo &bufferGroupInfo, QObject *parent = 0);
+	explicit AMDSClientContinuousDataRequest(ResponseType responseType, const QString &socketKey, const QString &bufferName, bool includeStatusData, const quint64 updateInterval, const QString &handShakeSocketKey, const AMDSBufferGroupInfo &bufferGroupInfo, QObject *parent = 0);
 	virtual ~AMDSClientContinuousDataRequest();
 	/// Copy constructor
 	AMDSClientContinuousDataRequest(const AMDSClientContinuousDataRequest &other);
 
 	/// Returns the start time for the data request
 	inline QDateTime startTime() const { return startTime_; }
+	/// Returns the update interval
+	inline quint64 updateInterval() const { return updateInterval_; }
+	/// Returns the last fetch time for the data request
+	inline QDateTime lastFetchTime() const { return lastFetchTime_; }
+	/// return the handShaking socket key
+	inline QString handShakingSocketKey() { return handShakeSocketKey_; }
+	/// check if it is hand shaking message
+	inline bool isHandShakingMessage() { return handShakeSocketKey_.length() > 0; }
+	/// Returns the last handshake time for the data request
+	inline QDateTime lastHandShakingTime() const { return lastHandShakeTime_; }
+	/// Returns whether message expired (didn't receive handshake in a given time interval
+	bool isExpired();
 
 	AMDSClientContinuousDataRequest& operator =(const AMDSClientContinuousDataRequest &other);
 
+	/// Return whether this is the original request message
+	inline bool isOriginalRequest() { return lastFetchTime_ == QDateTime(); }
```

Comment:
I think 
```C++
return !lastFetchTime_.isValid();
```
might better express intention and not require construction of another QDateTime to check against.